src/app/pages/videos/videos-list/services/videoslist-api.service.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { VideosList } from '../../../../models/videosList';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class VideoslistApiService {
+
+    constructor(private http: HttpClient) {}
+  
+    getVideoslist(
+      params?: Record<string, string>
+    ): Observable<VideosList[]> {
+      const newParams = this.createObjectToParams(params);
```

Comment:
const newParams = new HttpParams({ fromObject: params }).toString();

This will replace the entire function